lib/travis/api/app/endpoint/repos.rb

```diff
@@ -16,8 +16,13 @@ class Repos < Endpoint
       # json(:repositories)
       get '/' do
         prefer_follower do
-          params['ids'] = params['ids'].split(',') if params['ids'].respond_to?(:split)
-          respond_with service(:find_repos, params)
+          find_repos_service = service(:find_repos, params).run
+          if find_repos_service
+            params['ids'] = params['ids'].split(',') if params['ids'].respond_to?(:split)
```

Comment:
I hope it's okay if I leave a review -- IMO the intent would be clearer if the `current_user` check was here in the endpoint. And perhaps a more specific error than 404 could be returned to the client. For example a 403.

The case I have in mind: Someone is writing an API client and they are testing out the /repos endpoint but it returns a 404 despite the docs clearly saying it exists.